prov/rxd/src/rxd_av.c

```diff
@@ -185,25 +142,48 @@ static int rxd_av_insert(struct fid_av *av_fid, const void *addr, size_t count,
 	}
 
 	for (; i < count; i++, addr = (uint8_t *) addr + av->dg_addrlen) {
-		node = ofi_rbmap_find(&av->rbmap, (void *) addr);
-		if (node) {
-			rxd_addr = (fi_addr_t) node->data;
-		} else {
-			ret = rxd_av_insert_dg_addr(av, addr, &rxd_addr,
-						    flags, context);
-			if (ret)
-				break;
-		}
-
-		util_addr = av->rxd_addr_table[rxd_addr].fi_addr == FI_ADDR_UNSPEC ?
-			    rxd_set_fi_addr(av, rxd_addr) :
-			    av->rxd_addr_table[rxd_addr].fi_addr;
+		
+        ret = rxd_av_insert_dg_addr(av, addr, &dg_addr, flags, context);
+        if (ret)
+            break;
+        ret = rxd_av_insert_fi_addr(av, &dg_addr, &util_addr , flags);
+        if (ret) {
+            fi_av_remove(av->dg_av, &dg_addr, 1, flags);
+            break;    
+        }
+        dlist_foreach(&av->util_av.ep_list, av_entry) {
+
+            util_ep = container_of(av_entry, struct util_ep, av_entry);
```

Comment:
If we're only using the util_ep variable to get the rxd_ep, you should be able to just add the util_ep to the field names in the container ie
`rxd_ep = container_of(av_entry, struct rxd_ep, util_ep.av_entry);`